test/v2-migration-tests/src/test/resources/software/amazon/awssdk/v2migrationtests/maven/after/src/main/java/foo/bar/S3.java

```diff
@@ -384,4 +388,18 @@ private void setBucketNameTest(S3Client s3, String bucket) {
         GetObjectRequest getObjectRequest = GetObjectRequest.builder().bucket(bucket).key("key").bucket(bucket)
             .build();
     }
+
+    private void s3Uri(URI uri, String uriAsString) {
+        S3Uri s3Uri = S3Utilities.builder().build().parseUri(uri);
+
+        String versionId = s3Uri.firstMatchingRawQueryParameter("versionId").orElse(null);
+        String bucket = s3Uri.bucket().orElse(null);
+        String key = s3Uri.key().orElse(null);
+        String region = s3Uri.region().map(Region::id).orElse(null);
+        boolean isPathStyle = s3Uri.isPathStyle();
+
+        S3Uri s3UriFromString = /*AWS SDK for Java v2 migration: v2 S3Uri does not URL-encode a String URI. If you relied on this functionality in v1 you must update your code to manually encode the String.*/S3Utilities.builder().build().parseUri(URI.create(uriAsString));
```

Comment:
the Comment is kind of in a weird spot, is that what we do elsewhere for similar warning in migration tool?

Reply:
Yeah Dongie had same concern. We can't maintain the spacing if put on separate line though. Also for this specific transform can't do it on the line before since we transform the left and right hand sides separately. We could possibly do it on the next line. On the other hand, having it here makes it really obvious to the user

We do the same for other comments fwiw
https://github.com/aws/aws-sdk-java-v2/blob/master/v2-migration/src/test/java/software/amazon/awssdk/v2migration/AddCommentToMethodTest.java#L51

Reply:
Yeah, it's great that it is really obvious, just a bit weird. I don't mind it.